src/puya/awst_build/validation/arc4_copy.py

```diff
@@ -23,7 +23,7 @@ def visit_assignment_statement(self, statement: awst_nodes.AssignmentStatement)
 
     def _check_for_arc4_copy(self, expr: awst_nodes.Expression) -> None:
         match expr.wtype:
-            case wtypes.ARC4Array() | wtypes.ARC4Struct():
+            case wtypes.ARC4Array(immutable=False) | wtypes.ARC4Struct(immutable=False):
```

Comment:
looks like the `WType`s for these need updating to actually be made to be `immutable=False`?

Reply:
They already are set to False

Reply:
Ah right - in the factories, but not in the types themselves. Sorry, not sure how I missed that.

But it would be safer if we add
`immutable: bool = attrs.field(default=False, init=False)`
to the base wtype of the arrays `wtypes.ARC4Array`.